src/Adapter/MSTest.TestAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -249,9 +272,38 @@ internal TestCase ToTestCase()
             idProvider.AppendString(testCase.DisplayName);
```

Comment:
Any reason not to ditch the `display name`? It makes no sense to me to have "display name" as part of an ID. I would expect our ID to be simply similar to what roslyn does.

Reply:
It's there just to keep behavior consistent with 2.2.4+

Reply:
2.2.4 to 2.2.10 is using "display name" strategy, right? The "data" strategy is the new one you introduced to fix open bugs, right?

Reply:
Yep. You are correct. I will add some tests this weekend from my new flat hopefully.

Reply:
FYI I am actually writing tests now.